modules/build/src/main/scala/scala/build/preprocessing/directives/UsingScalaJsOptionsDirectiveHandler.scala

```diff
@@ -0,0 +1,102 @@
+package scala.build.preprocessing.directives
+
+import shapeless._
+
+import scala.build.Positioned
+import scala.build.errors.BuildException
+import scala.build.options.BuildOptions
+import scala.build.preprocessing.ScopePath
+import scala.util.{Success, Try}
+
+case object UsingScalaJsOptionsDirectiveHandler extends UsingDirectiveHandler {
+  def handle(directive: Directive, cwd: ScopePath): Option[Either[BuildException, BuildOptions]] =
+    ???
+
+  def name: String = "Scala JS options"
+
+  def description: String = "Add Scala JS options"
+
+  def usage: String = s"// using ${keys.mkString("|")} _value_"
+
+  override def usageMd: String =
+    """
+      |`// using jsVersion `_value_
+      |`// using jsMode `_value_
+      |`// using jsModuleKind `_value_
+      |`// using jsCheckIr true|false`
+      |`// using jsEmitSourceMaps true|false``
+      |`// using jsDom true|false``
+      |""".stripMargin
+
+  override def examples: Seq[String] = Seq(
+    "// using jsDom \"true\""
```

Comment:
The option the most likely to be used, and which makes the most sense to me here, is `moduleKind`.